spec/unit/type/file_spec.rb

```diff
@@ -1598,7 +1598,10 @@
       catalog.apply
 
       expect(Puppet::FileSystem.exist?(path)).to be_truthy
-      expect(@logs).not_to be_any {|l| l.level != :notice }
+      expect(@logs).not_to be_any { |l|
+        # the audit metaparameter is deprecated and logs a warning
+        l.level != :notice && !(/deprecated/.match(l.message))
+      }
```

Comment:
What do you think about adding a spec that verifies a deprecation warning is logged? 

Per prior discussion, we'll probably want an assertion that a deprecation warning is logged in strict mode, and an assertion that none is logged in "non-strict" mode.